### PR TITLE
Add button to delete all port devices

### PR DESCRIPTION
**Pull request by KraPete**
https://wirenboard.youtrack.cloud/issue/SOFT-4362/Knopka-Udalit-vse-ustrojstva

Screenshot_20241119_125552

## Release Notes

- **New Features**
	- Added a confirmation prompt for deleting all devices connected to a specific port.
	- Introduced buttons for deleting ports and devices, enhancing device management functionality.
	- Added a button to delete all port devices in the `wb-mqtt-serial` configuration.

- **Bug Fixes**
	- Improved error messages and notices during firmware updates for better user guidance.

- **Style**
	- Enhanced layout and responsiveness of the device management interface through CSS updates.

- **Documentation**
	- Updated changelog to reflect new features and improvements in device management.

### CODE REVIEW

File: app/scripts/react-directives/device-manager/config-editor/portTab.jsx [hunk 1]
```diff
@@ -19,21 +19,30 @@ export const PortTab = observer(({ tab }) => {
   );
 });
 
-export const PortTabContent = ({ tab, index, onDeleteTab }) => {
+export const PortTabContent = ({ tab, index, onDeleteTab, onDeletePortDevices }) => {
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/portTab.jsx:22]: Useless

File: app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js [hunk 2]
```diff
@@ -370,12 +370,32 @@
...
+      this.tabs.selectedTab?.name
+    );
+  }
+
+  async showDeletePortDevicesConfirmModal(portTab) {
+    return this.showDeleteConfirmModal(
+      'device-manager.labels.confirm-delete-port-devices',
+      portTab?.name
+    );
+  }
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js:385]: Useless

File: app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js [hunk 3]
```diff
@@ -370,12 +370,32 @@
     );
   }
 
+  async showDeleteTabConfirmModal() {
+    return this.showDeleteConfirmModal(
+      'device-manager.labels.confirm-delete',
+      this.tabs.selectedTab?.name
+    );
+  }
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js:378]: Useless

File: app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js [hunk 4]
```diff
@@ -370,12 +370,32 @@ class ConfigEditorPageStore {
...
+      'device-manager.labels.confirm-delete-port-devices',
+      portTab?.name
+    );
+  }
+
   async deleteTab() {
-    if ((await this.showDeleteConfirmModal()) == 'ok') {
+    if ((await this.showDeleteTabConfirmModal()) == 'ok') {
       this.tabs.deleteSelectedTab();
     }
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js:390]: Useless

File: app/scripts/react-directives/device-manager/config-editor/configEditorPage.jsx [hunk 5]
```diff
@@ -286,6 +292,7 @@ const ConfigEditorPage = observer(({ pageStore, onAddWbDevice, onSearchDisconnec
             selectedIndex={pageStore.tabs.selectedTabIndex}
             onSelect={(index, lastIndex) => pageStore.tabs.onSelectTab(index, lastIndex)}
             onDeleteTab={() => pageStore.deleteTab()}
+            onDeletePortDevices={tab => pageStore.deletePortDevices(tab)}
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/configEditorPage.jsx:295]: Useless

File: app/scripts/react-directives/device-manager/config-editor/tabsStore.js [hunk 6]
```diff
@@ -199,15 +199,11 @@
     const tab = this.items[this.selectedTabIndex];
     switch (tab?.type) {
       case TabType.PORT: {
-        tab.children.forEach(deviceTab => {
-          deviceTab.beforeDelete();
-          this.uniqueMqttIdChecker.removeTab(deviceTab);
-        });
-        this.items.splice(this.selectedTabIndex, tab.children.length + 1);
+        this.deleteDeviceTabs(tab);
+        this.items.splice(this.selectedTabIndex, 1);
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/tabsStore.js]: Done

File: app/scripts/react-directives/device-manager/config-editor/portTabStore.js [hunk 7]
```diff
@@ -112,6 +112,8 @@ export class PortTab {
   }
 
   deleteChildren(index) {
+    const deviceTab = this.children.at(index);
+    deviceTab?.beforeDelete();
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/portTabStore.js:116]: Useless

File: app/scripts/react-directives/device-manager/config-editor/tabsStore.js [hunk 8]
```diff
@@ -306,4 +302,25 @@ export class TabsStore {
...
+      return;
+    }
+    for (let i = 0; i < deviceTabsCount; i++) {
+      const deviceTab = portTab.children[0];
+      portTab.deleteChildren(0);
+      this.uniqueMqttIdChecker.removeTab(deviceTab);
+    }
+    this.items.splice(portTabIndex + 1, deviceTabsCount);
+    this.hasModifiedStructure = true;
+  }
```
[the remark this answers is not in the document]
Comment (KraPete) [reply app/scripts/react-directives/device-manager/config-editor/tabsStore.js:325]: Useless

File: app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ninedev-i) [reply app/scripts/react-directives/device-manager/config-editor/configEditorPageStore.js:390]: ну, я согласен вообще с coderabbit :)

File: app/scripts/react-directives/device-manager/config-editor/tabsStore.js [hunk 9]
```diff
@@ -306,4 +302,25 @@ export class TabsStore {
       return false;
     });
   }
+
+  deletePortDevices(portTab) {
```
Comment (ninedev-i) [inline app/scripts/react-directives/device-manager/config-editor/tabsStore.js:306]: Здесь переусложнено, но я не могу предложить хорошего решения :(

**Review by ninedev-i — APPROVED** [review]

**Pull request merged**